ruby/basics/arrays.md

```diff
@@ -34,28 +49,105 @@ Array.new(3, Array.new) #=> [[], [], []]
 
-Calling `#methods` on an array will also yield a long list of methods, like so:
+Calling `#methods` on an array will also yield a long list of available methods, like so:
 
 ```
-x = [1, 2, 3, 4, 5]
-x.methods               #=> A very long list of methods
+num_array.methods       #=> A very long list of methods
+```
 
-* Accessing via index
-* Push/pop/shovel
-* Iteration: map/collect <!-- I'd leave enumerables to its own lesson-->
-* Addition/concatenation and substraction/difference
-* Other useful methods (empty?, include?, reverse, join)
+### Accessing element
+Like most other programming languages, Ruby arrays use **zero-based indexing**. As you learned in Chris Pine's tutorial, accessing an array's element is as simple as calling the `#[]` method with the **index**, or position, you desire. Recall that calling an invalid position will result in `nil`. Ruby also allows the use of negative indices, which return elements starting from the *end* of an array at [-1].
 
-## Exercises
-A group of exercises (If Applicable) for the student to complete in relation to the topic taught in the lesson.
+```
+str_array = ["This", "is", "a", "small", "array"]
+
+str_array[-1]           #=> "array"
+str_array[-2]           #=> "small"
+```
+
+Finally, Ruby provides the `#first` and `#last` methods, which should be self-explanatory. What may not be obvious, however, is that these methods can take an argument that will return a *new* array.
+
+```
+str_array = ["This", "is", "a", "small", "array"]
+
+str_array.first         #=> "This"
+str_array.first(2)      #=> ["This", "is"]
+```
+
+### Adding and removing elements
+Adding an element to an existing array is as simple at calling `#push` or the shovel operator `<<`. Both methods will add elements to the end of an array and return that array, so the methods can be chained. The `#pop` method will remove an element from the end of an array and return that element, rather than returning the array. Therefore, it cannot be chained.
+
+```
+num_array = [1, 2]
+
+num_array.push(3, 4)      #=> [1, 2, 3, 4]
+num_array.push(5).push(6) #=> [1, 2, 3, 4, 5, 6]
+
+num_array << 7            #=> [1, 2, 3, 4, 5, 6, 7]
+
+num_array.pop             #=> 7
+num_array.pop.pop         #=> undefined method 'pop' for Fixnum
+```
+
+The methods `#shift` and `#unshift` can also be used to operate at the beginning of an array. While `#shift` will remove the first element of an array and return that element (much like `#pop`), `#unshift` will add elements to the beginning of an array and return that array.
+
+```
+[1, 2, 3, 4].unshift(0)   #=> [0, 1, 2, 3, 4]
+
+[1, 2, 3, 4].shift        #=> [1]
+[1, 2, 3, 4].shift(2)     #=> [1, 2]
+```
+
+### Adding and subtracting arrays
+What do you think will be the outcome of `[1, 2, 3] + [3, 4, 5]`?
+
+If you guessed `[1, 2, 3, 3, 4, 5]`, congratulations! The `#+` method will return a new array built by concatenating the two arrays. The `#concat` method works the same way.
```

Comment:
same issue as the brackets. this might be true, but it can be confusing to a newbie to refer to the `#+` method. Just say that "Adding the two arrays with return a new array built by concatening...~